### PR TITLE
Replace pgbuild.sh with a Makefile

### PR DESCRIPTION
**Pull request by hlinnaka**
This allows building both Zenith and PostgreSQL in one command. The
command is 'make'

### CODE REVIEW

**Comment by arssher** [comment]
I've pushed a couple of things; `cargo test -- --test-threads=1` still fails for me because basebackup seems to go nowhere (`pg_ctl: directory "/home/ars/zenith/zenith/control_plane/../tmp_check/pgbench/pgdatadirs/pg1" is not a database cluster directory`), but evidently this is independent of the makefile/pgbuild.sh stuff as it still fails with pgbuild.sh anyway. Will look at it a bit later.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	$(MAKE) -C tmp_install/build/ MAKELEVEL=0 clean
+
+# This doesn't remove the effects of 'configure'.
+clean:
+	cd tmp_install/build && ${MAKE} clean
+	cargo clean
+
+# This removes everything
+distclean:
+	rm -rf tmp_install
```
Comment (lubennikovaav) [inline Makefile:41]: Shouldn't we add `cd tmp_install/build && ${MAKE} distclean` here?

Comment (hlinnaka) [reply Makefile:41]: Not necessary when we do `rm -r tmp_install`. That removes the whole thing anyway.

**Comment by hlinnaka** [comment]
> I've pushed a couple of things; `cargo test -- --test-threads=1` still fails for me because basebackup seems to go nowhere (`pg_ctl: directory "/home/ars/zenith/zenith/control_plane/../tmp_check/pgbench/pgdatadirs/pg1" is not a database cluster directory`), but evidently this is independent of the makefile/pgbuild.sh stuff as it still fails with pgbuild.sh anyway. Will look at it a bit later.

Thanks! I don't think the `-j8` belongs in the makefile, it will be passed through if you do `make -j8` on the command line. (I habitually use `make -j4 -s` myself`). I squashed the other fix, I'll push shortly if the test pass.

**Comment by hlinnaka** [comment]
> > 
> 
> Thanks! I don't think the `-j8` belongs in the makefile, it will be passed through if you do `make -j8` on the command line. (I habitually use `make -j4 -s` myself`). I squashed the other fix, I'll push shortly if the test pass.

On the other hand, `cargo build` implicitly uses multiple parallel jobs. And the output actually doesn't look nice if you pass `make -j` on the command line, as that makes the Postgres build and cargo build run in parallel, and that messes up the graphical output from cargo build. Maybe we should force `cargo build` and `postgres` to not run in parallel, by adding `.NOTPARALLEL:`

**Comment by arssher** [comment]
Indeed so. LGTM. I would probably add
```
# cargo build is parallel on its own (recursive invocation of postgres target
# still gets '-j' from above passed)
```
above .NOTPARALLEL, but this is not absolutely necessary.

Also, shouldn't we replace `cargo build` in testing.yml with `make zenith`?

**Comment by hlinnaka** [comment]
> Also, shouldn't we replace cargo build in testing.yml with make zenith? (root install pushes headers as well, so this doesn't matter much either)

Hmm. The github action caches the 'postgres' build, so that it doesn't need to be recompiled every time; that's why I didn't change testing.yml it to do just 'make`. But you're right that it could still use `make zenith`.

**Comment by hlinnaka** [comment]
> > Also, shouldn't we replace cargo build in testing.yml with make zenith? (root install pushes headers as well, so this doesn't matter much either)
> 
> Hmm. The github action caches the 'postgres' build, so that it doesn't need to be recompiled every time; that's why I didn't change testing.yml it to do just 'make`. But you're right that it could still use `make zenith`.

I tried that, and it didn't work. It tried to run 'configure' and install the Postgres headers, which failed. I'm not sure what's going on there, and perhaps it could be fixed some other way, but I'll go with `cargo build` for now.

**Pull request merged**